M/MLX_C/build_tarballs.jl

```diff
@@ -0,0 +1,78 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "MLX_C"
+version = v"0.0.11" # This is an unofficial version number - used for git commit sha (126253...) matching MLX v0.20.0
+
+sources = [
+    GitSource("https://github.com/ml-explore/mlx-c.git", "126253627b95a0803273f44655fdaf5e7be4fbe1"),
+    ArchiveSource("https://github.com/roblabla/MacOSX-SDKs/releases/download/macosx14.0/MacOSX14.0.sdk.tar.xz",
+                  "4a31565fd2644d1aec23da3829977f83632a20985561a2038e198681e7e7bf49"),
+    DirectorySource("./bundled"),
+]
+
+script = raw"""
+# MLX hack
+cd $includedir
+mv mlx/io.h mlx/io.h.bak
+cp mlx/io.h.bak mlx/io.h
```

Comment:
What?

Reply:
He he - yeah, it’s a hackish workaround for the MLX headers needing a patch - and it seemed like `patch` didn’t like patching the (targets of the) symlinks pointing to the MLX headers in the Julia artifacts (patch required the files to be normal files).

The needed patch is already upstreamed, but post MLX v0.21.0.

Reply:
I really don't understand what's going on here.  A comment would be most helpful to avoid questions.  Also, I'm totally missing why you need to do `mv` + `cp` instead of just `cp`.

Reply:
Luckily superfluous now, with MLX v0.21.0 (which include the std::unordered_map patch).

... but what I was doing was:
1. Rename the symlink to `mlx/io.h.bak` to free up `mlx/io.h` for patching.
2. Copy the (content of the) symlink from `mlx/io.h.bak` to `mlx/io.h`
3. Patch `mlx/io.h`

Reply:
(I'll try to remember a reasonably verbose comment next time)